LambdaLang.Tests/OptionTests.cs

```diff
@@ -0,0 +1,61 @@
+using SemanticAlgebra;
+using SemanticAlgebra.Option;
+using Xunit;
+
+namespace LambdaLang.Tests;
+
+public class OptionTests
+{
+    [Fact]
+    public void Select_WithSome_ReturnsTransformedValue()
+    {
+        // Arrange
+        var s = Option.Some(40);
+        
+        // Act
+        var e = s.Select(n => n + 2);
+        
+        // Assert
+        // Verify that e is Option.Some containing 42
+        var result = e.Evaluate(new OptionExtractSemantic<int>());
```

Comment:
Just verify it with Option.Some(42) equals, no need to use extract semantic, similar for other related code